docs/install/dns-setup.md

```diff
@@ -0,0 +1,156 @@
+sudo systemctl enable dnsmasq.service
+sudo service dnsmasq restart
+sudo echo "DNS=127.0.0.1" >> /etc/systemd/resolved.conf
+sudo echo "DOMAINS=~example.com" >> /etc/systemd/resolved.conf
+sudo service systemd-resolved restart
+```
+
+#### Windows
+
+Unfortunately dnsmasq will not run on Windows and I have not found something similar yet.
+
+#### MacOS
+
+On MacOS you can alias a private IP address to the loop back interface e.g. `10.128.0.1` with `sudo ifconfig lo0 alias 10.128.0.1`
+
+You will need install dnsmasq using (homebrew)[https://docs.brew.sh/Installation]
+
+```
+brew install dnsmasq
+```
+
+It appears that the install location differs based on the Apple Hardware. For Intel hardware Macs it's in `/usr/local` and for M1 hardware it's `/opt/homebrew`. Please check where it installed things when the previous command has completed.
+
+Then edit a configuration file 
+
+M1 mac
+```
+echo "conf-dir=/opt/homebrew/etc/dnsmasq.d" >> /opt/homebrew/etc/dnsmasq.conf
+echo "address=/example.com/10.128.0.1" > /opt/homebrew/etc/dnsmasq.d/ff.conf
+```
+
+Intel mac
+```
+echo "conf-dir=/usr/local/etc/dnsmasq.d" >> /usr/local/etc/dnsmasq.conf
+echo "address=/example.com/10.128.0.1" > /usr/local/etc/dnsmasq.d/ff.conf
+```
+
+Set dnsmasq to run as a service
+
+```
+brew services start dnsmasq
+dscacheutil -flushcache
+```
+
+Tell MacOS to use dnsmasq for our test domain
+
+```
+sudo mkdir -p /etc/resolver
+sudo tee /etc/resolver/example.com > /dev/null <<EOF
+nameserver 127.0.0.1
+domain example.com
+search_order 1
+EOF
+```
+
+And finally kick the MacOS resolver so it sees the updates
+
+```
+sudo killall -HUP mDNSResponder
+```
+
+### Pi Hole
+
+Pi Hole is a package that bundles dnsmasq as an image to run on a Raspberry Pi (or in Docker container e.g. on your NAS). Its main use is to block advertisements embedded in web pages. But since in its normal configuration it is already handling all the local DNS traffic, making us work with FlowForge is possible and means you do not need to change any settings on your development/test machine.
+
+Create the following file in `/etc/dnsmasq.d` called `02-flowforge.conf`
+
+```
+address=/example.com/192.168.0.22
+```
+
+Where 192.168.0.22 is the ipv4 address of the Docker host machine or a Kubernetes node. And `example.com` is the domain to use.
+
+After making the change you will probably need to restart things with:
+
+```
+sudo pihole restartdns
+```
+
+If running Pi Hole in Docker then you will need to create the file on the host and mount it to the `/etc/dnsmasq.d/02-flowforge.conf` location.
+
+### No Local DNS server
+
+If you really can't run dnsmasq then there is a possible alternative.
+
+You can set the `domain` entry in the `flowforge.yml` configuration file to the following pattern `172.17.0.1.sslip.io`
+
+This works because the `sslip.io` domain is set up to always return the IP address embedded in the hostname queried.
```

Comment:
```suggestion
### No Local DNS server

If you really can't run dnsmasq then there is a possible alternative.

A really useful service called sslip.io allows you to test FlowForge even when you cannot use dnsmasq. You can read more about [sslip.io on their web site](https://sslip.io).

You will need to set the `domain` entry in the `flowforge.yml` configuration file to the following pattern `172.17.0.1.sslip.io`. Don't forget to replace the IP address with the correct one for your configuration.

This will work because the `sslip.io` domain is set up to always return the IP address embedded in the hostname queried. 
```